core/src/main/java/com/linecorp/armeria/internal/common/metric/RequestMetricSupport.java

```diff
@@ -66,6 +68,8 @@ public static void setup(
                           RequestLogProperty.NAME,
                           RequestLogProperty.SESSION)
            .thenAccept(log -> onRequest(log, meterIdPrefixFunction, server, successFunction));
+
+        MoreMeters.setDistributionStatisticConfig(distributionStatisticConfig);
```

Comment:
Instead of your advice(https://github.com/line/armeria/pull/4829#discussion_r1188050037), I've override a distributionStatisticConfig from here.

Do you think this is too dangerous approach?
@minwoox

Reply:
It is not thread safe. The two requests having different `DistributionStatisticConfig`s update this the global `DistributionStatisticConfig` concurrently, the result is unpredictable.

Reply:
That makes sense to me 😞

Reply:
As @ikhoon said, we can't just set the config here.
Is there any reason that you are reluctant to use the config in the `onResponse` method?

Reply:
> Let's pass it to the `setup` method in `execute(...)` so we can use it when creating the distribution:
> 
> ```
> RequestMetricSupport.setup(ctx, REQUEST_METRICS_SET, meterIdPrefixFunction, false,
>                            successFunction != null ? successFunction::test
>                                                    : ctx.options().successFunction(),
>                            distributionStatisticConfig);
> ```
> 
> In `RequestMetricSupport`
> 
> ```
> private static void onRequest(
>         RequestLog log, MeterIdPrefixFunction meterIdPrefixFunction, boolean server,
>         SuccessFunction successFunction, DistributionStatisticConfig distStatCfg) { ...}
> 
> private static void onResponse(
>         RequestLog log, MeterIdPrefixFunction meterIdPrefixFunction, boolean server,
>         SuccessFunction successFunction, DistributionStatisticConfig distStatCfg) {
>         ...
>         new DefaultClientRequestMetrics(distStatCfg);
>         ...
>         new DefaultServiceRequestMetrics(distStatCfg);
>         ...
>     }
> ```

@minwoox I looked into `DefaultClientRequestMetrics` class but it seems like there is no injection point as of now.
Because the constructor of the class calls `MoreMeters.newTimer()` but to pass given `distStatCfg` I need to change current implementation of `newTimer()`.
That's why I tried to use setter of `MoreMeters`.

If you implied for me to modify `newTimer` method signature and its logic, just let me know. I will do it :)